File: solutions/typescript/VB_T5_002_greeter_io_boundary.ts
// VB-T5-002: Greeter IO Boundary -- TypeScript baseline
function buildGreeting(name: string): string { return "Hello, " + name + "!\n"; }
function greetIO(name: string): void { process.stdout.write(buildGreeting(name)); }
console.assert(buildGreeting("Alice") === "Hello, Alice!\n");

File: solutions/python/VB_T5_002_greeter_io_boundary.py
"""VB-T5-002: Greeter IO Boundary -- Python baseline."""
def build_greeting(name: str) -> str:
    return "Hello, " + name + "!\n"

def greet(name: str) -> None:
    print(build_greeting(name), end="")

if __name__ == "__main__":
    assert build_greeting("Alice") == "Hello, Alice!\n"
    print("VB-T5-002 ok")
